packages/paste-core/components/timeline/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,141 @@
+      Event details
+    </TimelineItem>
+
+    <TimelineItem
+      title="Item without timestamp"
+      collapsible
+      collapsibleHeading="custom heading"
+      data-testid="timelineItemCollapsibleNoTimestamp"
+    >
+      Event details
+    </TimelineItem>
+
+    <TimelineItem
+      title="Item with timestamp and collapsibleHeading"
+      timestamp="2018-03-01:10:00"
+      collapsible
+      collapsibleHeading="custom heading"
+      data-testid="timelineItemCollapsibleHeading"
+    >
+      Event details
+    </TimelineItem>
+  </Timeline>
+);
+
+describe("Timeline", () => {
+  it("should render", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timeline")).toBeDefined();
+    expect(getByTestId("timelineItem")).toBeDefined();
+  });
+
+  it("should display title and timestamp", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItem")).toHaveTextContent("Start");
+    expect(getByTestId("timelineItem")).toHaveTextContent("2018-03-01:10:00");
+  });
+
+  it("should display content", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItem")).toHaveTextContent("Event details");
+  });
+
+  it("should display item without timestamp", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(
+      getByTestId("timelineItemNoTimestamp").querySelector("[data-paste-element='TIMELINE_ITEM_TIMESTAMP']"),
+    ).toBeNull();
+  });
+
+  it("should display icon item", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItem").querySelector("[data-paste-element='TIMELINE_ITEM_ICON']")).toBeDefined();
+  });
+
+  it("should display grouped items", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItemGroup")).toBeDefined();
+  });
+
+  it("should display group item timestamp", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(
+      getByTestId("timelineItem").querySelector("[data-paste-element='TIMELINE_ITEM_GROUP_TIMESTAMP_DETAIL_TEXT']"),
+    ).toBeDefined();
+  });
+
+  it("should display collapsible items", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItemCollapsible")).toBeDefined();
+  });
+
+  it("should display custom collapsible heading", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItemCollapsibleNoTimestamp")).toHaveTextContent("custom heading");
+  });
+
+  it("should not display custom collapsible heading text when timestamp is p", () => {
+    const { getByTestId } = render(<ExampleTimeline />);
+    expect(getByTestId("timelineItemCollapsibleHeading")).toHaveTextContent("2018-03-01:10:00");
+  });
+
+  describe("Customization", () => {
```

Comment:
suggestion(blocking): let's add TimelineItemIcon to these tests and any other elements we attach the data-paste-element attribute to